MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -476,6 +480,39 @@ void EnggDiffFittingPresenter::enableMultiRun(
 
 void EnggDiffFittingPresenter::processStart() {}
 
+void EnggDiffFittingPresenter::processLoad() {
+  // while file text-area is not empty
+  // while directory vector is not empty
+  // if loaded here set a global variable true so doesnt load again?
```

Comment:
I would avoid global/status variables. To optimize this, when a new run number is given we can check if that is the run number of the  current/already loaded `engggui_fitting_focused_ws`. I think we only keep one / the last one, so this wouldn't be a big priority.